tools/build/scanCode.py

```diff
@@ -117,96 +177,133 @@ def run_line_checks(file_path):
+
 def all_paths(root_dir):
+    """Generator that returns files with known extensions that can be scanned.
+
+    Iteration is recursive beginning at the passed root directory and
+    skipping directories that are listed as exception paths.
+    """
     for dir_path, dir_names, files in os.walk(root_dir):
         for f in files:
             path = os.path.join(dir_path, f)
             if all(map(lambda p: not path.endswith(p), exceptional_paths())):
                 yield os.path.join(dir_path, f)
 
-# Helpers, rather than non-standard modules.
+
 def colors():
+    """Create a collection of helper functions to colorize strings."""
     ansi = hasattr(sys.stderr, "isatty") and platform.system() != "Windows"
 
     def colorize(code, string):
         return "%s%s%s" % (code, string, '\033[0m') if ansi else string
 
-    blue  = lambda s: colorize('\033[94m', s)
-    green = lambda s: colorize('\033[92m', s)
-    red   = lambda s: colorize('\033[91m', s)
+    def blue(s):
+        return colorize('\033[94m', s)
 
-    return collections.namedtuple("Colorizer", "blue green red")(blue, green, red)
+    def green(s):
+        return colorize('\033[92m', s)
+
+    def red(s):
+        return colorize('\033[91m', s)
+
+    return collections.namedtuple("Colorizer",
+                                  "blue green red")(blue, green, red)
 
 # Script entrypoint.
 if __name__ == "__main__":
-    if len(sys.argv) != 2:
-        sys.stderr.write("Usage: %s root_directory.\n" % sys.argv[0])
+
+    # Prepare message colorization methods
+    col = colors()
+
+    # Test necessary arguments exist
+    if len(sys.argv) < 2:
+        sys.stderr.write(col.red(MSG_SCRIPT_USAGE % sys.argv[0]))
         sys.exit(1)
 
     root_dir = sys.argv[1]
 
-    col = colors()
-
     if not os.path.isdir(root_dir):
-        sys.stderr.write("%s: %s is not a directory.\n" % (sys.argv[0], root_dir))
+        sys.stderr.write(ERR_PATH_IS_NOT_DIRECTORY %
+                         (sys.argv[0], root_dir))
 
     # This determines which checks run on which files.
     file_checks = [
-        ("*",          [ is_not_symlink ]),
-        ("*.scala",    [ has_block_copyright, line_checks([ no_tabs, no_trailing_spaces, eol_at_eof ]) ]),
-        ("*.py",       [ line_checks([ no_tabs, no_trailing_spaces, eol_at_eof ]) ]),
-        ("*.java",     [ has_block_copyright, line_checks([ no_tabs, no_trailing_spaces, eol_at_eof ]) ]),
-        ("*.js",       [ line_checks([ no_tabs, no_trailing_spaces, eol_at_eof ]) ]),
-        ("build.xml",  [ line_checks([ no_tabs, no_trailing_spaces, eol_at_eof ]) ]),
-        ("deploy.xml", [ line_checks([ no_tabs, no_trailing_spaces, eol_at_eof ]) ]),
-        ("*.gradle",   [ line_checks([ no_tabs, no_trailing_spaces, eol_at_eof ]) ]),
-        ("*.md",       [ line_checks([ no_tabs, eol_at_eof ]) ]),
-        ("*.go",       [ has_block_copyright, line_checks([ no_tabs, no_trailing_spaces, eol_at_eof ]) ])
+        ("*", [is_not_symlink]),
```

Comment:
"should"? too strong.

there are other coding conventions not just Pep8. i think when editing pre-existing code, it's better to follow the conventions it already applies when there are no enforcement in place.  
we can discuss this elsewhere as this could get into the weeds.